src/GafferUIModule/PathListingWidgetBinding.cpp

```diff
@@ -1175,8 +1018,8 @@ class PathModel : public QAbstractItemModel
 						QVariant decorationData;
 						try
 						{
-							displayData = model->m_columns[i]->data( path, Qt::DisplayRole, canceller );
-							decorationData = model->m_columns[i]->data( path, Qt::DecorationRole, canceller );
+							displayData = valueToVariant( *model->m_columns[i], path, GafferUI::PathColumn::Role::Value, canceller );
+							decorationData = valueToVariant( *model->m_columns[i], path, GafferUI::PathColumn::Role::Icon, canceller );
```

Comment:
This makes more sense after the later commit where you add a lot more options here, and store in a vector instead of two hardcoded variables.

Reply:
I think this is an interesting question even with the later commits. It's definitely true that Value, Background and Icon will always be needed together, no matter the strategy taken in PathModel, because they are all needed to draw a cell. In theory, for some uses, there may be a benefit in deferring Tooltip evaluation, but in practice we definitely want to evaluate it at the same time in the background process, so it's ready to go and won't hang the UI if the query is slow later. And when it comes to hooking this up in the LightEditor, it's definitely mildly inconvenient that everything is queried separately. That all points to a single return value holding everything perhaps being preferable.

Technically it would be an ABI break to add a value later even with the current approach, because adding an enum value breaks ABI. But we have been doing that regularly for TypeIds and getting away with it, and it seemed defensible that we might choose to do the same here. And there is a fairly likely addition on the horizon, with an Annotation that we could use for doing the `+-/*=` annotation for tweaks in the LightEditor.

I don't think I have a particularly good reason for it, but I am still kinda leaning towards doing it this way, at least for now. One to talk about when we catch up later?

Reply:
Tentative decision :

- Return a new `Cell` struct with everything in it, rather than returning thing one-by-one
- Keep the fields in the struct as `RunTimeTyped/Data` for now, but consider making them more strongly typed later
- Reserve enough space to add a couple more fields in future

Reply:
Implemented this in https://github.com/GafferHQ/gaffer/pull/4501/commits/732a59e0dd871f32972df66cf8ba552f11c0b1c9. I've also rebased to fix some merge conflicts, but that is the only new commit. Would be great to get this merged and then I can rebase my LightEditor branch on top of it.